### PR TITLE
API/OP (ConditionalBlock) error message enhancement

### PR DESCRIPTION
**Pull request by zhhsplendid**
Enhance Python `ConditionalBlock` API input type check and C++ `ConditionalBlock` PADDLE_ENFORCE checks

### CODE REVIEW

File: paddle/fluid/operators/controlflow/conditional_block_op_helper.cc [hunk 1]
```diff
@@ -45,9 +48,10 @@ static void FindAllConditionalBlockAndConditionalBlockGradOp(
     }
   }
 
-  PADDLE_ENFORCE_GE(
-      fwd_ops->size(), bwd_ops->size(),
-      "There are extra conditional_block_grad ops in the graph or program");
+  PADDLE_ENFORCE_GE(fwd_ops->size(), bwd_ops->size(),
+                    platform::errors::InvalidArgument(
+                        "There are more conditional_block_grad ops then "
+                        "conditional_block ops in the graph or program"));
```
Comment (Aurelius84) [inline paddle/fluid/operators/controlflow/conditional_block_op_helper.cc]: `then` -> `than`?

File: paddle/fluid/operators/controlflow/conditional_block_op_helper.cc [hunk 2]
```diff
@@ -31,7 +31,10 @@ static bool IsMatchedConditionalBlockOpAndConditionalBlockGradOp(
 static void FindAllConditionalBlockAndConditionalBlockGradOp(
     const framework::ProgramDesc &program, std::vector<OpVariant> *fwd_ops,
     std::vector<OpVariant> *bwd_ops) {
-  PADDLE_ENFORCE_GE(fwd_ops->size(), bwd_ops->size());
+  PADDLE_ENFORCE_GE(
+      fwd_ops->size(), bwd_ops->size(),
+      platform::errors::InvalidArgument(
+          "Size of forward ops must be greater or equal to backward ops"));
```
Comment (liym27) [inline paddle/fluid/operators/controlflow/conditional_block_op_helper.cc]: May be better to print `fwd_ops->size()` and `fwd_ops->size()`.

File: paddle/fluid/operators/controlflow/conditional_block_op.cc [hunk 3]
```diff
@@ -56,7 +56,9 @@ class ConditionalBlockOp : public ConditionalOp {
 
     if (need_run) {
       auto *scope_var = scope.FindVar(Output(ConditionalOp::kScope));
-      PADDLE_ENFORCE(scope_var != nullptr, "Must set scope");
+      PADDLE_ENFORCE_NE(scope_var, nullptr,
+                        platform::errors::InvalidArgument(
+                            "Scope must be set in conditional_block_op"));
```
Comment (liym27) [inline paddle/fluid/operators/controlflow/conditional_block_op.cc]: May be better to use `PADDLE_ENFORCE_NOT_NULL`  and `platform::errors::PreconditionNotMet`.

File: paddle/fluid/operators/controlflow/conditional_block_infer_op.cc [hunk 4]
```diff
@@ -51,7 +51,9 @@ class ConditionalBlockInferOp : public ConditionalOp {
 
     if (need_run) {
       auto *scope_var = scope.FindVar(Output("Scope"));
-      PADDLE_ENFORCE(scope_var != nullptr, "Must set scope");
+      PADDLE_ENFORCE_NE(scope_var, nullptr,
+                        platform::errors::InvalidArgument(
+                            "Scope must be set in ConditionalBlockInferOp"));
```
Comment (liym27) [inline paddle/fluid/operators/controlflow/conditional_block_infer_op.cc]: May be better to use `PADDLE_ENFORCE_NOT_NULL`  and `platform::errors::PreconditionNotMet`.

File: paddle/fluid/operators/controlflow/conditional_block_op_helper.cc (hunk 1 above)
Comment (zhhsplendid) [reply paddle/fluid/operators/controlflow/conditional_block_op_helper.cc]: Done

File: paddle/fluid/operators/controlflow/conditional_block_op_helper.cc (hunk 2 above)
Comment (zhhsplendid) [reply paddle/fluid/operators/controlflow/conditional_block_op_helper.cc]: Done

File: paddle/fluid/operators/controlflow/conditional_block_op.cc (hunk 3 above)
Comment (zhhsplendid) [reply paddle/fluid/operators/controlflow/conditional_block_op.cc]: Done

File: paddle/fluid/operators/controlflow/conditional_block_infer_op.cc (hunk 4 above)
Comment (zhhsplendid) [reply paddle/fluid/operators/controlflow/conditional_block_infer_op.cc]: Done

**Review by liym27 — APPROVED** [review]
LGTM

**Review by Aurelius84 — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**